source/manual/alerts/email-alert-api-app-healthcheck-not-ok.html.md

```diff
@@ -4,28 +4,85 @@ title: email-alert-api app healthcheck not ok
+last_reviewed_on: 2019-12-03
 review_in: 6 months
 ---
 
 If there is a health check error showing for Email Alert API, you can click on
-the alert to find out more details about what’s wrong. Here are the possible
-problems you may see:
+the alert to find out more details about what’s wrong. We've listed some common
+problems and remedies later on in this document.
 
-Some of the alerts below tend to trigger when emails are sent out. Check to see
-if emails have sent out by looking at the
-[Email Alert API Metrics Grafana dashboard][dashboard].
+## You may not need to do anything
+
+Some alerts tend to trigger when emails are sent out. Check to see if emails have
+sent out by looking at the [Email Alert API Metrics Grafana dashboard][dashboard].
 
 This affects the following alerts:
 
 * Unprocessed content changes
 * Subscription content
 * High queue alerts (all)
-* Subscription content
 
 In this case you can wait until the emails have all been sent out.
 
-## Unprocessed content changes (content_changes)
+## General troubleshooting tips
+
+### Check that Notify sent the email
+
+Log in to the Notify account by going to <https://www.notifications.service.gov.uk>.
+The login credentials are in the [2nd line password store][password-store] under
+`govuk-notify/2nd-line-support`. Two-factor authentication links for logging in are
+sent to the 2nd line email address.
+
+From the dashboard, you can search by email address.
+
+### Check the courtesy copy inbox
+
+A courtesy copy of every email sent from integration, staging and production
+is available in the ['govuk-email-courtesy-copies' Google Group][google-group].
+Integration and staging emails have a subject prefixed by the
+environment name.
+
+If you need to test the process of subscribing to an email alert, you
+can subscribe with `govuk-email-courtesy-copies@digital.cabinet-office.gov.uk`.
+You should then see two copies of the email at that address when you
+perform an action that triggers an email alert (the courtesy copy and
+the subscriber one).
+
+The process for triggering an alert varies by publishing application,
+but it usually involves creating a new edition of some content, then
+publishing it with a 'major' update type. The change note you enter
+will appear in the email that gets sent.
+
+> **Note**
+>
+> Due to the way that we send the courtesy copy emails and the
+> anonymisation integration data sync, we can't fully exercise the code
+> that matches content changes with subscriptions. Therefore, it is not
+> safe to rely on the courtesy copy Google Group to guarantee that emails
+> are being sent to subscribers, and instead a better test is to
+> subscribe yourself and check that emails arrive.
+
+## Check the email alert API
+
+A useful debug step if you're not sure if an email has been sent is to
+check Email Alert API to see if it has created records in its database.
+
+```
+$ govuk_app_console email-alert-api
+> ContentChange.last
+> Email.last
+> DeliveryAttempt.last
```

Comment:
Email records (and I think the others too) have a uuid as a primary key so `Email.last` won't necessarily return the most recently created email, you might be better off with something like `Email.where(created_at: DateTime.now - 15.minutes...DateTime.now).count` or something

Reply:
Thanks for the review! I agree this is more of a sanity check than a truly reliable test. For example, if another email is coincidentally created around the same time as you're debugging you could get some confusing results!

Not sure how changing to `created_at` would make the sanity check any better, unless I'm missing something 🤔`uuid` should always return the most recently created email, whereas `created_at` can be overridden when the email is created so that it appears to be older than it is.

Reply:
Sorry, I wasn't clear. `.last` finds the record that comes last according to the database primary key. For most apps where a strictly increasing integer is used, that works perfectly. However, for any app that uses a uuid as the primary key, the order will be alphabetical so it won't return the most recent record but the one with the 'highest' alphabetical order, which is unlikely to be the most recently created. Agreed, someone could come along and alter the created_at fields but I think it's a bit unlikely

Reply:
Ahhh I'm with you. Makes sense! I can edit the doc to that effect.